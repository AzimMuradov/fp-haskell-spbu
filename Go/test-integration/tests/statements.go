func main() {
  var n = 5;


  /* if - else */

  // should print "5"
  if n == 3 + 2 {
    println(n);
  };

  // should print nothing
  if n == 73 {
    println(n);
  };

  // should print "42"
  if n != 3 + 2 {
    println(n);
  } else {
    println(42);
  };

  // should print "25"
  if n == 3 + 3 {
    println(n);
  } else if n == 6 - 1 {
    println(n * n);
  };


  /* for */

  i := 1;

  for i := 0; i < 17; i++ {
    println(i * i * i);
  };

  println(i);

  i := 2;
  for i < 5 {
    println(i);
    i++;
  };
  println(i);

  for {
    println("true");
    break;
  };
}
