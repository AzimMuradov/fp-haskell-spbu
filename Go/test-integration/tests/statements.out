5
42
25
0
1
8
27
64
125
216
343
512
729
1000
1331
1728
2197
2744
3375
4096
1
2
3
4
5
true
